# 7.2 (center of mass of a sphere)

In [ ]:
using LinearAlgebra
using Plots
using Colors
using Statistics
using Polynomials
gr()

We can solve this problem using different differential elements, but since this is an excersize on multivariable integrals, we will use the spherical coordinates and random values for $r$, $\phi$ and $\theta$ and $\rho(r, \theta) = \frac{\rho_{max}}{2}(\frac{3}{2} + \frac{r \cos(\theta)}{2 R})$ where $\rho_{max}$ is the density at the highest point of the sphere and $R$ is the radius of the sphere. We could have taken advantage of the cylindrical symmetries of the problem to reduce the problem to a single variable integral over $z$ but that is not the focus for now.

Therefore the Integral we should take to calculate the height of the center of mass from the center of the sphere is as follows:<br>
$z_{cm} = \frac{1}{M} \int_0^R \int_0^{2 \pi} \int_0^\pi r^3 \cos(\theta) \frac{\rho_{max}}{2}(\frac{3}{2} + \frac{r \cos(\theta)}{2 R})dr \sin(\theta) d\phi d\theta $ <br>
where <br>
$M = \int_0^R \int_0^{2 \pi} \int_0^\pi \frac{\rho_{max}}{2}(\frac{3}{2} + \frac{r \cos(\theta)}{2 R}) r^2 dr \sin(\theta) d\phi d\theta $

We won't define a seperate function this time but we will use the same function as the simpleIntegrate function used in the previous problem to calculate this integral.

Our ranged random generator function:

In [ ]:
function randRanged(lowRange, upRange)
    y = lowRange + (upRange - lowRange) * rand()
    return y
end

Integration parameters: (for simplicity $\rho_{max}, R = 1$ and the integral over $\phi$  is already taken)

In [37]:
rhoMax = 1
R = 1
fz(r, theta) = 2 * pi * r^3 * sin(theta) * cos(theta) * rhoMax / 2 * (3 / 2 + r * cos(theta) / 2 /R)  #function that we want to integrate for z
fm(r, theta) = 2 * pi * r^2 * sin(theta) * rhoMax / 2 * (3 / 2 + r * cos(theta) / 2 /R)    #function that we want to integrate for M
lowerBoundR = 0 #lower boundary for r
upperBoundR = R #upper boundary for r
lowerBoundTheta = 0 #lower boundary for theta
upperBoundTheta = pi #upper boundary for theta
intRes = 10000000 #resolution of the integral (number of samples taken)

#Now for the Integration
fzSamples = Float64[]
for i in 1:intRes
    push!(fzSamples, fz(randRanged(lowerBoundR, upperBoundR), randRanged(lowerBoundTheta, upperBoundTheta)))
end
Iz = (upperBoundR - lowerBoundR) * (upperBoundTheta - lowerBoundTheta) * mean(fzSamples)

fmSamples = Float64[]
for i in 1:intRes
    push!(fmSamples, fm(randRanged(lowerBoundR, upperBoundR), randRanged(lowerBoundTheta, upperBoundTheta)))
end
Im = (upperBoundR - lowerBoundR) * (upperBoundTheta - lowerBoundTheta) * mean(fmSamples)

zCm = Iz / Im
println("The center of mass is $zCm units above the center")

The center of mass is 0.06655744911620101 units above the center


So the center of mass is approximately 0.06656 units above the center of the sphere for $\rho_{max}, R = 1$.